### PR TITLE
Fix reported failure on ESM update

### PR DESCRIPTION
**Pull request by superm1**
I received a private report on ESM update failures occurring on Panamera with master.  I believe it's a logic error in the update path for ESM.

Make some adjustments to more closely mirror the regular update path.
I have successfully upgraded/downgraded about 5 times with this change in place, so I do believe it's safe.

cc @ryanchang0624 for any comments

### CODE REVIEW

File: plugins/synapticsmst/synapticsmst-common.c [hunk 1]
```diff
@@ -470,7 +470,7 @@ synapticsmst_common_enable_remote_control (SynapticsMSTConnection *connection,
 							 UPDC_ENABLE_RC,
 							 5, 0, (guint8*)sc,
 							 &error_local)) {
-			g_warning ("Failed to enable remote control in layer %d: %s, retrying",
+			g_debug ("Failed to enable remote control in layer %d: %s, retrying",
 				   i, error_local->message);
```
Comment (hughsie) [inline plugins/synapticsmst/synapticsmst-common.c]: indent needs fixing too...

**Comment by hughsie** [comment]
No complaints from me, I don't understand the hardware enough to NAK/ACK this kind of thing.

File: plugins/synapticsmst/synapticsmst-common.c (hunk 1 above)
Comment (superm1) [reply plugins/synapticsmst/synapticsmst-common.c]: fixed

**Comment by superm1** [comment]
> No complaints from me, I don't understand the hardware enough to NAK/ACK this kind of thing.

OK.  Functionally the only difference is timing, which waiting longer between commands should do nothing but help failures. I'll merge this for now, and if Ryan indicates that the time to sleep after erase should be shorter, can follow up with a 1 line diff later.

**Comment by ryanchang0624** [comment]
> > No complaints from me, I don't understand the hardware enough to NAK/ACK this kind of thing.
> 
> OK. Functionally the only difference is timing, which waiting longer between commands should do nothing but help failures. I'll merge this for now, and if Ryan indicates that the time to sleep after erase should be shorter, can follow up with a 1 line diff later.

Hi Mario,
I saw you add a 5s delay for erase failure, right?  We retry if erase failed but don't have this kind of delay within. Maybe it is related the flash itself, so I have no concern about the delay but shorter is better if no compatibility issue.

**Comment by superm1** [comment]
Well added two things specifically.
1. Delay after erase
2. Redo erase if ESM write failed before rewriting ESM.

It wasn't clear to me from the error report which part actually failed, but I conjectured that they are both possible failures.  
1. The hypothesis for the first failure was that if SPI wasn't fully erased by the time write started some blocks might be `0x00` instead of `0xFF` which would cause the checksum to be invalid.
2. The hypothesis for the second failure was that if the write failed some of the blocks might be invalid and the only way to correct them would be to erase and try again.

So although this will take longer for the write process, it seems much safer for those hypothesis at least.

**Pull request merged**